bindata/assets/alerts/cpu-utilization-sno.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: cpu-utilization
+  namespace: openshift-kube-apiserver
+spec:
+  groups:
+    - name: control-plane-cpu-utilization
+      rules:
+        - alert: HighOverallControlPlaneCPU
+          annotations:
+            summary: >-
+              CPU utilization across plane pods is high; a control plane pods outage may cause a cascading failure;
+              check control plane logs for possible issues and/or add more CPU and modify alert threshold.
+            runbook_url: PASS
+            description: >-
+              By default, this alert is triggered if all control plane pods are using more that 3 CPUs(3000 millicores).
+              Usually, when they start to utilize more CPU, this means that there are might be issues at them.
+              It's recommended to check pods and their logs to verify that there are not issues.
+              If all is ok, then increase the CPU on the node and increase alert threshold.
+          expr: |
+            sum(
+              rate(
+                container_cpu_system_seconds_total{namespace=~"openshift-etcd|openshift-kube-apiserver|openshift-kube-controller-manager|openshift-kube-scheduler|openshift-apiserver|openshift-controller-manager|openshift-authentication",image!=""
+                }[5m]
+              )
+            )
+            > 3
```

Comment:
This is assuming that the number of vCPUs allocated for the platform components is fixed for all setups?

Reply:
> it can be always modified

how? the value is hardcoded in the expression so you'll change for all or nothing?

Reply:
The primary use case for `AlertingRule` is for the cluster admins to setup new alerting rules that aren't covered by the product.
For out-of-the-box alerting rules, it's preferred to have adaptive thresholds if possible.

Reply:
I thought that it was possible to pin the management workload to a subset of the cores. In which case sum(machine_cpu_cores) can be larger than the number of cores allocated for control plane components?

https://docs.openshift.com/container-platform/4.14/scalability_and_performance/enabling-workload-partitioning.html
https://docs.openshift.com/container-platform/4.14/scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.html#ztp-sno-du-enabling-workload-partitioning_sno-configure-for-vdu

Reply:
But when it's enabled, is there a way to know how many cores are allocated?

Reply:
Ideally you want to build an alert expression that takes into account the variability of workload partitioning and another metric reporting the number of cores being allocated is the way to go indeed. Having said, I'm not a maintainer of this repository so I don't feel strongly whether you should do it now or later.